src/transaction_sender/mock_transfer_generator.zig

```diff
@@ -1,52 +1,167 @@
+        }),
+    };
+};
+
 pub const MockTransferService = struct {
     allocator: std.mem.Allocator,
     sender: *Channel(TransactionInfo),
+    rpc_client: RpcClient,
     exit: *AtomicBool,
+    logger: Logger,
+    accounts: MockAccounts = MockAccounts.DEFAULT,
 
-    pub fn init(allocator: std.mem.Allocator, sender: *Channel(TransactionInfo), exit: *AtomicBool) !MockTransferService {
+    pub fn init(
+        allocator: std.mem.Allocator,
+        sender: *Channel(TransactionInfo),
+        rpc_client: RpcClient,
+        exit: *AtomicBool,
+        logger: Logger,
+    ) !MockTransferService {
         return .{
             .allocator = allocator,
             .sender = sender,
+            .rpc_client = rpc_client,
             .exit = exit,
+            .logger = logger,
         };
     }
 
-    /// Mock transaction generator that sends a transaction every 10 seconds
-    /// Used to test the transaction sender
-    /// TODO:
-    /// - Pass sender keypair and receiver pubkey
-    pub fn run(self: *MockTransferService) !void {
-        errdefer self.exit.store(true, .unordered);
-
-        const from_pubkey = try Pubkey.fromString("Bkd9xbHF7JgwXmEib6uU3y582WaPWWiasPxzMesiBwWm");
-        const from_keypair = KeyPair{
-            .public_key = .{ .bytes = from_pubkey.data },
-            .secret_key = .{ .bytes = [_]u8{ 76, 196, 192, 17, 40, 245, 120, 49, 64, 133, 213, 227, 12, 42, 183, 70, 235, 64, 235, 96, 246, 205, 78, 13, 173, 111, 254, 96, 210, 208, 121, 240, 159, 193, 185, 89, 227, 77, 234, 91, 232, 234, 253, 119, 162, 105, 200, 227, 123, 90, 111, 105, 72, 53, 60, 147, 76, 154, 44, 72, 29, 165, 2, 246 } },
-        };
-        const to_pubkey = try Pubkey.fromString("GDFVa3uYXDcNhcNk8A4v28VeF4wcMn8mauZNwVWbpcN");
-        const lamports: u64 = 100;
+    /// Run the mock transfer service
+    pub fn run(self: *MockTransferService, n_transactions: u64, n_lamports_per_tx: u64) !void {
+        errdefer self.exit.store(true, .monotonic);
+
+        var prng = std.Random.DefaultPrng.init(19);
+        const random = prng.random();
+
+        self.logger.info().logf("(transaction_sender.MockTransferService) bank: {s}, alice: {s}", .{
+            self.accounts.bank.pubkey,
+            self.accounts.alice.pubkey,
+        });
+
+        const total_bank_balance = n_transactions * n_lamports_per_tx;
+        self.logger.info().logf("(transaction_sender.MockTransferService): : {} txs each transfers {} lamports", .{ n_transactions, n_lamports_per_tx });
+        const balances = try self.logBalances("(transaction_sender.MockTransferService) starting");
+        if (balances.bank < total_bank_balance) {
+            self.logger.info().log("(transaction_sender.MockTransferService) airdropping to bank");
+            try self.airdrop(self.accounts.bank.pubkey, total_bank_balance);
```

Comment:
Since we can get limited on airdrops, I'd just set the requested airdrop to something high like 5 sol, rather than the required balance for 1 run